src/web-animations-next-animation.js

```diff
@@ -25,13 +27,48 @@
     this._animation = null;
     this._childAnimations = [];
     this._callback = null;
+    this._oldPlayState = 'idle';
     this._rebuildUnderlyingAnimation();
     // Animations are constructed in the idle state.
     this._animation.cancel();
+    this._updatePromises();
   };
 
   scope.Animation.prototype = {
+    _updatePromises: function() {
+      var oldPlayState = this._oldPlayState;
+      var newPlayState = this.playState;
+      if (this._readyPromise && newPlayState !== oldPlayState) {
+        if (newPlayState == 'idle') {
+          if (this._readyPromiseState == 'pending') {
+            this._rejectReadyPromise();
+          }
+          this._resetReadyPromise();
+          this._resolveReadyPromise();
+        } else if (oldPlayState == 'pending') {
+          this._resolveReadyPromise();
+        } else if (newPlayState == 'pending') {
+          this._resetReadyPromise();
+        }
+      }
+      if (this._finishedPromise && newPlayState !== oldPlayState) {
+        if (newPlayState == 'idle') {
+          if (this._finishedPromiseState == 'pending') {
+            this._rejectFinishedPromise();
+          }
+          this._resetFinishedPromise();
+        } else if (newPlayState == 'finished') {
+          this._resolveFinishedPromise();
+        } else if (oldPlayState == 'finished') {
+          this._resetFinishedPromise();
+        }
+      }
+      if (this._oldPlayState !== this.playState) {
```

Comment:
> this if is redundant?

Yeah. There used to be a method that changed the oldPlayState, now it's just an assignment there is no point checking.